### DISCUSSION TITLE
Eliminate "Create a new store" page?

### DISCUSSION THREAD

**Discussion by bolatovumar** [btcpayserver/btcpayserver#2685] [Ideas / Feature Requests]
Since we require a name for a store before it's created we have a special page for it which the user sees when they want to create a store. I wonder if we could eliminate that page entirely and just give a store a random name like "Untitled Store" like Google Docs does for new documents or come up with random names so that the store is created right away and you can avoid an extra click and some typing. Users can always come up with an appropriate name later.

![Screen Shot 2021-07-08 at 3 22 54 PM](https://user-images.githubusercontent.com/1934678/124998284-fb8ee580-e000-11eb-9d8a-026730c87015.png)

**Comment by Zaxounette** [comment]
Discussed, I think in [Design Call N°12](https://github.com/btcpayserver/organization/issues/42).
Don't see it in my notes, so @dennisreimann or @dstrukt might be able to refresh my memory on the consensus that was found on this.

**Reply by dennisreimann** [reply]
I think we didn't discuss this one in particular, but the intermediate app creation form, which ist similar. 

However, @NicolasDorier brought this up [on Mattermost](https://chat.btcpayserver.org/btcpayserver/pl/scoe8wmai3r37fs9uq1uysh4te) recently too. 

These are my 2 sats from the conversation:

> I just checked whether or not there'd be more options to present there on the initial view. Even though imho there aren't I still think it makes sense to name the store immediately, because you'll most likely want to do that anyways. This way you'd have at least a clear distinction on the list in case you have multiple stores.

As we are in the process of overhauling the creation views, I'm sure @dstrukt has some thoughts here as well.

**Reply by dstrukt** [reply]
There are a few considerations here.

1. Once the store name is randomly generated or untitled named, if the user wants to immediately rename it, they must step through a few additional clicks to make this happen, and wouldn't (at least with our current implementation) function like a Google Doc, as nice as that functionality is. This would end up adding more steps to something that was otherwise intended to shorten the creation process.

2. This generative/rename later pattern of creation/editing is not used anywhere else in the app.

3. I think we intended to add more to this configuration screen in the future, or we're enabled to do so, in the current implementation.

It's possible there's more to it than the above points from a technical perspective, but from a UX / design perspective, those are my two sats.

**Reply by dennisreimann** [reply]
Agreed!

**Reply by bolatovumar** [reply]
Ok, if we are planning on adding more configuration options to this screen in the future then we can keep it.

**Reply by dstrukt** [reply]
Definitely worth discussing what other things could/should be surfaced here. 

Will make a note of this for the next design meeting and see if there's anything that makes sense at this time.

**Comment by pavlenex** [comment]
Closing this one as tracker is now here https://github.com/btcpayserver/btcpayserver/issues/6728
